kinds/gen.go

```diff
@@ -182,3 +172,23 @@ func die(err error) {
 	fmt.Fprint(os.Stderr, err, "\n")
 	os.Exit(1)
 }
+
+func compileProvider(fsys fs.FS, rt *thema.Runtime) (*kindsys.Provider, error) {
```

Comment:
We should probably have a function like this in kindsys

Reply:
plausible - let's work it out here first, then we can upstream it